src/DynamoCore/Core/PulseMaker.cs

```diff
@@ -0,0 +1,133 @@
+﻿using System;
+using System.Threading;
+
+using Dynamo.Models;
+
+namespace Dynamo.Core
+{
+    public class PulseMaker
+    {
+        #region Class Data Members and Properties
+
+        private readonly Timer internalTimer;
+        private readonly object stateMutex;
+        private SynchronizationContext context;
+        private bool evaluationInProgress = false;
+        private bool evaluationRequestPending = false;
+        internal int TimerPeriod { get; private set; }
+
+        public event Action<SynchronizationContext> RunStarted;
+        protected virtual void OnRunStarted(SynchronizationContext context)
```

Comment:
It's strictly not `PulseMaker`'s responsibility to provide a `SynchronizationContext` to other classes.